main.c

```diff
@@ -37,124 +50,45 @@ int main() {
         switch (state) {
             case 0: {
                 // Input personal data
```

Comment:
if the code is written well, comments should not be necessary; what you could do for this switch for example is define constants for the steps (#define INPUT_PERSONAL_DATA_STEP 0 and so on), making these comments at each step useless